db/index.js

```diff
@@ -1,32 +1,64 @@
 'use strict';
 
-function createDBQuery(spec) {
-    const {logger} = spec;
-
-    // TODO: Integrate with ORACLE
-
-    // // https://node-postgres.com/guides/project-structure
-    // async function query(text, params) {
-    //     const hrstart = process.hrtime();
-    //     const results = await pool.query(text, params);
-    //     const hrend = process.hrtime(hrstart);
-    //     const executionTime = {
-    //         seconds: hrend[0],
-    //         milliseconds: hrend[1] / 1000000
-    //     };
-    //     const duration = `${executionTime.seconds}s ${executionTime.milliseconds}ms`;
-
-    //     logger.info(
-    //         {query: {text, duration, executionTime, rows: results.rowCount}},
-    //         'DB QUERY EXECUTED'
-    //     );
-
-    //     return results;
-    // }
-
-    // return Object.freeze({
-    //     query
-    // });
+const oracledb = require('oracledb');
+
+oracledb.initOracleClient({libDir: '/opt/oracle/instantclient_21_10'});
+const getSecret = require('../services/secret-manager/index');
+const logger = require('../services/logging/logger');
+
+// Generates an insert statment for tarriff
+function generateInsertStatement(jsonObject, table) {
+    let columnsList = '';
+    let columnsValues = '';
+    Object.keys(jsonObject).forEach(column => {
+        columnsList = `${columnsList + column}, `;
+        columnsValues = `${columnsValues}:${column}, `;
+    });
+    columnsList = columnsList.slice(0, -2);
+    columnsValues = columnsValues.slice(0, -2);
+
+    const statement = `INSERT INTO ${table} (${columnsList}) VALUES (${columnsValues})`;
+
+    return statement;
+}
+
+async function createDBConnection(oracleObject) {
```

Comment:
Probably some structural changes to make down the line for this function, things like having the createDbConneciton function only responsible for creating the connection and having a re-usable (singleton?) client that we can run INSERT statements against separately